### PR TITLE
[CDEC-485] Move Block data types from core to chain package

### PR DESCRIPTION
**Pull request by ruhatch**
## Description

We want to reduce the size of the `core` package and put everything related to blocks and chains in the `chain` package. This first PR clears the way for that and move the `Pos.Core.Block` modules across.

## Linked issue

[CDEC-485](https://iohk.myjetbrains.com/youtrack/issue/CDEC-485)

## Type of change

- [~] 🐞 Bug fix (non-breaking change which fixes an issue)
- [~] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [~] 🔨 New or improved tests for existing code
- [~] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [~] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/cardano-sl-core.cabal [hunk 1]
```diff
@@ -82,15 +74,6 @@ library
...
                        Pos.Core.Binary.Orphans
 
-                       -- Block
-                       Pos.Core.Block.Genesis.Lens
-                       Pos.Core.Block.Genesis.Instances
-                       Pos.Core.Block.Genesis.Types
-                       Pos.Core.Block.Main.Instances
-                       Pos.Core.Block.Main.Types
-                       Pos.Core.Block.Union.ComponentBlock
-                       Pos.Core.Block.Union.Types
```
Comment (mhuesch) [inline core/cardano-sl-core.cabal]: 👋

File: core/test/Test/Pos/Core/Bi.hs [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+       , exampleTxPayload
+       , exampleTxProof
+       , exampleUpdatePayload
+       , exampleUpdateProof
+       , staticHeavyDlgIndexes
+       , staticProxySKHeavys
+
+       -- Helpers
        , feedPM
+       , feedPMC
```
Comment (mhuesch) [inline core/test/Test/Pos/Core/Bi.hs]: ^ 👍

File: chain/src/Pos/Chain/Block/JsonLog.hs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+module Pos.Chain.Block.JsonLog
+       ( jlCreatedBlock
+       , jlAdoptedBlock
+       ) where
+
+import           Formatting (sformat)
```
Comment (intricate) [inline chain/src/Pos/Chain/Block/JsonLog.hs]: Not a huge deal, but:
```
...
import           Universum

import           Formatting
...
```

File: chain/src/Pos/Chain/Block/Main/Types.hs [hunk 4]
```diff
@@ -54,7 +54,7 @@ instance Bi MainProof where
                          decode <*>
                          decode
 
-instance SafeCopy SscProof => SafeCopy MainProof where
```
Comment (intricate) [inline chain/src/Pos/Chain/Block/Main/Types.hs]: Was this a redundant constraint or something? Just curious.

Comment (parsonsmatt) [reply chain/src/Pos/Chain/Block/Main/Types.hs]: The old code base used these to side step orphan instance problems.

**Pull request merged**